test/integration/targets/ec2_snapshot/tasks/main.yml

```diff
@@ -0,0 +1,250 @@
+#      register: result
+#    - assert:
+#        that:
+#          - result is not changed
+
+    - name: Take snapshot if most recent >1hr (False)
+      ec2_snapshot:
+        volume_id: '{{ volume_id }}'
+        last_snapshot_min_age: 60
+      register: result
+
+    - name: Get info about snapshots
+      ec2_snapshot_info:
+        filters:
+          "tag:Name": '{{ resource_prefix }}'
+      register: info_result
+
+    - assert:
+        that:
+          - result is not changed
+          - info_result.snapshots| length == 1
+
+    - name: Pause so we can do a last_snapshot_min_age test
+      pause:
+        minutes: 1
+
+#    JR: Check mode not supported
+#    - name: Take snapshot if most recent >1min (True) (check mode)
+#      ec2_snapshot:
+#        volume_id: '{{ volume_id }}'
+#        snapshot_tags:
+#          Name: '{{ resource_prefix }}'
+#        last_snapshot_min_age: 1
+#        check_mode: true
+#      register: result
+#    - assert:
+#        that:
+#          - result is changed
+
+    - name: Take snapshot if most recent >1min (True)
+      ec2_snapshot:
+        volume_id: '{{ volume_id }}'
+        last_snapshot_min_age: 1
+      register: result
+
+    - name: Get info about snapshots
+      ec2_snapshot_info:
+        filters:
+          "tag:Name": '{{ resource_prefix }}'
+      register: info_result
+
+    - assert:
+        that:
+          - result is changed
+          - info_result.snapshots| length == 2
+          - '"{{ result.snapshot_id }}" in "{{ info_result| json_query("snapshots[].snapshot_id") }}"'
+
+#    JR: Check mode not supported
+#    - name: Take snapshot with a tag (check mode)
+#      ec2_snapshot:
+#        volume_id: '{{ volume_id }}'
+#        snapshot_tags:
+#          MyTag: '{{ resource_prefix }}'
+#      register: result
+#    - assert:
+#        that:
+#          - result is changed
+
+    #  Wait at least 15 seconds between concurrent volume snapshots.
+    - name: Prevent SnapshotCreationPerVolumeRateExceeded errors
+      pause:
+        seconds: 15
+
+    - name: Take snapshot and tag it
+      ec2_snapshot:
+        volume_id: '{{ volume_id }}'
+        snapshot_tags:
+          MyTag: '{{ resource_prefix }}'
+      register: result
+
+    - name: Get info about snapshots by tag
+      ec2_snapshot_info:
+        filters:
+          "tag:MyTag": '{{ resource_prefix }}'
+      register: tag_info_result
+
+    - assert:
+        that:
+          - result is changed
+          - result.tags| length == 2
+          - tag_info_result.snapshots| length == 1
+          - result.tags.MyTag == "{{ resource_prefix }}"
+          - '"{{ result.snapshot_id }}" in "{{ tag_info_result| json_query("snapshots[].snapshot_id") }}"'
+
+    - name: Get info about all snapshots for this test
+      ec2_snapshot_info:
+        filters:
+          "tag:Name": '{{ resource_prefix }}'
+      register: info_result
+
+    - assert:
+        that:
+          - info_result.snapshots| length == 3
+
```

Comment:
Prior to deleting the snapshot we should probably test that we can re-tag the snapshot.

The docs don't explicitly call it out, but if you pass a snapshot_id you can add tags to an existing snapshot.  (don't like that you can't remove them, but that's out of scope here)

Reply:
I'm not seeing how this would work, ec2.create_snapshot does not take a snapshot_id. Can you provide a sample task?

Reply:
Ah OK that works, but also doesn't correctly handle changed status.  I started to fix that and it quickly exploded so I'll PR that feature separately before coming back here. Thx.